### PR TITLE
Remove wrong resolveInput types until fixed

### PR DESCRIPTION
**Pull request by dcousens**
This pull request removes the contextualised type for `resolvedData` in `resolveInput`, and it's return type.

In a followup change, will will be adding contextualised types back to this method, but as is, the types are misleading in different situations and may be providing a false sense of security about the safety of your code.

### CODE REVIEW

File: packages/core/src/lib/core/types-for-lists.ts [hunk 1]
```diff
@@ -128,68 +128,89 @@ function getIsEnabled(listsConfig: KeystoneConfig['lists']) {
...
   { storage: configStorage, lists: listsConfig, db: { provider } }: KeystoneConfig,
   listGraphqlTypes: Record<string, ListGraphQLTypes>,
   intermediateLists: Record<string, { graphql: { isEnabled: IsEnabled } }>
 ) {
-  return Object.fromEntries(
-    Object.entries(listsConfig).map(([listKey, list]) => [
+  const result: Record<string, PartiallyInitialisedList> = {};
+
+  for (const listKey in listsConfig) {
+    const list = listsConfig[listKey];
```
Comment (mitchellhamilton) [inline packages/core/src/lib/core/types-for-lists.ts:141]:
```suggestion
  for (const [listKey, list] of Object.entries(listsConfig)) {
```

Comment (mitchellhamilton) [reply packages/core/src/lib/core/types-for-lists.ts:141]: (and the other similar things)

Comment (dcousens) [reply packages/core/src/lib/core/types-for-lists.ts]: will follow that up momentarily

File: examples/custom-field/3-pair-field/index.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  CommonFieldConfig<ListTypeInfo> & {
+    isIndexed?: boolean | 'unique';
+  };
+
+export function pair<ListTypeInfo extends BaseListTypeInfo>({
+  isIndexed,
+  ...config
+}: PairFieldConfig<ListTypeInfo> = {}): FieldTypeFunc<ListTypeInfo> {
+  function resolveInput(value: string | null | undefined) {
+    if (!value) return { left: null, right: null };
```
Comment (mitchellhamilton) [inline examples/custom-field/3-pair-field/index.ts:21]:
```suggestion
    if (!value) return { left: value, right: value };
```

File: examples/custom-field/3-pair-field/index.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          index: isIndexed === true ? 'index' : isIndexed || undefined,
+        },
+      },
+    })({
+      ...config,
+      input: {
+        where: {
+          arg: graphql.arg({ type: filters[meta.provider].String.optional }),
+          resolve: filters.resolveCommon,
+        },
```
Comment (mitchellhamilton) [inline examples/custom-field/3-pair-field/index.ts:58]:
```suggestion
```

File: examples/custom-field/3-pair-field/index.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+          arg: graphql.arg({
+            type: orderDirectionEnum,
+          }),
+          resolve(orderDirection, context) {
+            return {
+              left: orderDirection,
+              right: orderDirection,
+            };
+          },
+        },
```
Comment (mitchellhamilton) [inline examples/custom-field/3-pair-field/index.ts:84]:
```suggestion
```

File: examples/custom-field/3-pair-field/views.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        in: {
+          label: 'Is one of',
+          initialValue: '',
+        },
+        not_in: {
+          label: 'Is not one of',
+          initialValue: '',
+        },
+      },
+    },
```
Comment (mitchellhamilton) [inline examples/custom-field/3-pair-field/views.tsx:132]:
```suggestion
```

File: examples/custom-field/schema.ts [hunk 6]
```diff
@@ -1,29 +1,45 @@
...
+      //   but it isn't very meaningful in context
+      resolveInput: async ({ resolvedData }) => {
+        //          console.log('Post.hooks.resolveInput', { resolvedData, operation, inputData, item });
+        return {
+          ...resolvedData,
+
+          // add some defaults
+          pair: {
+            left: resolvedData?.pair?.left || '<blank>',
+            right: resolvedData?.pair?.right || '<blank>',
```
Comment (mitchellhamilton) [inline examples/custom-field/schema.ts:39]:
```suggestion
            left: resolvedData.pair?.left || '<blank>',
            right: resolvedData.pair?.right || '<blank>',
```

**Review by mitchellhamilton — APPROVED** [review]

**Pull request merged**